beard/clustering/tests/test_wrappers.py

```diff
@@ -41,20 +40,124 @@ def testData():
     return make_blobs(centers=4, shuffle=False, random_state=random_state)
 
 
-def test_unsupervised(testData):
-    """Unsupervised clustering."""
-    X, y = testData
+@pytest.fixture
+def unsupervised():
+    """Preparing a test data."""
+    random_state = check_random_state(42)
+
+    X, y = make_blobs(centers=4, shuffle=False, random_state=random_state)
+    return X
+
+
+@pytest.fixture
+def semiSupervised():
+    """Preparing a test data for semi_Supervised cases."""
+    random_state = check_random_state(42)
+    X, y = make_blobs(centers=4, shuffle=False, random_state=random_state)
+    mask = random_state.randint(2, size=len(y)).astype(np.bool)
+    y[mask] = -1
+    d = euclidean_distances(X)
+    d = (d + d.T) / 2.0
+    d /= d.max()
+    return d, y
+
+
+# Defining a scoring function for the following 2 test cases
+def _groundTruth_X_score(X, true_labels, pred_labels):
+    """Scoring function that takes exactly 3 arguments."""
+    # For the sake of the test cases, X will be ignored. So, it will be like
+    # using a function that takes exactly 2 arguments.
+    score = b3_f_score(true_labels, pred_labels)
+    return score
+
+
+def test_scipy_HAC_semi_supervised_labels_affinity(semiSupervised):
+    """Test semi-supervised learning for sp.HAC when scoring_data=affinity."""
+    X, y = semiSupervised
+
+    # We should find all 4 clusters
+    clusterer = ScipyHierarchicalClustering(scoring=_groundTruth_X_score,
+                                            scoring_data="affinity",
+                                            affinity=euclidean_distances)
+    clusterer.fit(X, y)
+    labels = clusterer.labels_
+    assert_array_equal([25, 25, 25, 25], np.bincount(labels))
+
+
+def test_scipy_HAC_semi_supervised_labels_raw(semiSupervised):
```

Comment:
Python naming convenions don't allow for names like `semiSupervised` (mixed case)

https://www.python.org/dev/peps/pep-0008/#prescriptive-naming-conventions

Rephrase it to `semi_supervised` or `semisupervised` everywhere where it appears.